<launch>

    <rosparam file="$(find pkg_ros_iot_bridge)/config/config_pyiot.yaml"/>

    <!-- Launch Task-5 Simulation Environment in Gazebo -->
    <include file="$(find pkg_vb_sim)/launch/task6_simulation.launch"/>

    <!-- Launch Moveit move_group Node -->
    <include file="$(find pkg_vb_sim)/launch/two_ur5_move_group.launch"/>

    <!-- Run Task-5 Pick Place ROS Nodes for UR5_1 and UR5_2 -->
    <node name="node_t5_ur5_1" pkg="pkg_task5"
          type="node_t5_ur5_1.py" output="screen"/>
    <node name="node_t5_ur5_2" pkg="pkg_task5"
          type="node_t5_ur5_2.py" output="screen"/>

    <node pkg="pkg_ros_iot_bridge"
          type="node_action_server_ros_iot_bridge.py"
          name="node_action_server_ros_iot_bridge"
          output="screen"/>

    <!--    RViz Scene Files for ur5_1 and ur5_2-->
    <group ns="ur5_1">
        <arg name="scene_file"
             default="$(find pkg_task5)/config/rviz/ur5_1.scene"/>
        <node name="moveit_publish_scene_from_text" pkg="moveit_ros_planning"
              type="moveit_publish_scene_from_text" args="$(arg scene_file)"/>
    </group>

    <group ns="ur5_2">
        <arg name="scene_file"
             default="$(find pkg_task5)/config/rviz/ur5_2.scene"/>
        <node name="moveit_publish_scene_from_text" pkg="moveit_ros_planning"
              type="moveit_publish_scene_from_text" args="$(arg scene_file)"/>
    </group>

    <!-- Recording Bag File for Submission -->
    <arg name="record" default="false"/>
    <arg name="rec_name" default="t5.bag"/>

    <group if="$(arg record)">
        <node name="rosbag_record_pick" pkg="rosbag" type="record"
              args="record -O $(find pkg_task5)/bag_files/$(arg rec_name) --chunksize=10 /eyrc/vb/ur5_1/vacuum_gripper/logical_camera/ur5_1 /eyrc/vb/ur5_2/vacuum_gripper/logical_camera/ur5_2"
              output="screen"/>
    </group>

</launch>